service/telemetry/logger.go

```diff
@@ -43,6 +131,15 @@ func newLogger(set Settings, cfg Config) (*zap.Logger, log.LoggerProvider, error
 	// To make sure they are also exposed in logs written to stdout, we add them as fields to the Zap core created above using WrapCore.
 	// We do NOT add them to the logger using With, because that would apply to all logs, even ones exported through the core that wraps
 	// the LoggerProvider, meaning that the attributes would be exported twice.
+	logger, lp := configureLogger(logger, cfg, set)
+
+	return logger, lp, nil
+}
+
+// configureLogger applies common configuration to the logger
+func configureLogger(logger *zap.Logger, cfg Config, set Settings) (*zap.Logger, log.LoggerProvider) {
+	// The attributes in cfg.Resource are added as resource attributes for logs exported through the LoggerProvider
+	// To make sure they are also exposed in logs written to stdout, we add them as fields to the Zap core
```

Comment:
If you want to split this to a function, please, keep the comment from lines 130 to 133 here. The reason is that this only applies to the statement behind but not to the others.